src/OpenTelemetry.Api/DistributedContext/DistributedContext.cs

```diff
@@ -116,10 +111,12 @@ public static DistributedContextCarrier Carrier
             {
                 if (value is null)
                 {
-                    throw new ArgumentNullException(nameof(value));
+                    OpenTelemetryApiEventSource.Log.InvalidArgument("set_Carrier", "value is null");
```

Comment:
in this case let's assign no-op carrier to the Carrier field. Perhaps that is what user meant by setting it to `null`